poppy/wfe.py

```diff
@@ -353,6 +353,151 @@ def get_opd(self, wave):
         phase_screen = np.fft.ifftshift(np.fft.ifft2(np.fft.ifftshift(scaled))).real   # FT of scaled random PSD makes phase screen
 
         phase_screen -= np.mean(phase_screen)  # force zero-mean
-        opd = phase_screen / np.std(phase_screen) * self.wfe.to(u.m).value  # normalize to wanted input rms wfe
+        self.opd = phase_screen / np.std(phase_screen) * self.wfe.to(u.m).value  # normalize to wanted input rms wfe
 
-        return opd
+        return self.opd
+
+
+class PowerSpectrumWFE(WavefrontError):
+    """
+    Power spectrum PSD WFE class from characterizing and modeling 
+    optical surface power spectrum and applying optical noise.
+    
+    References:
+    Males, Jared. MagAO-X Preliminary-Design Review, 
+        Section 5.1: Optics Specifications, Eqn 1
+        https://magao-x.org/docs/handbook/appendices/pdr/
+    Lumbres, et al. In Prep.
+
+    Parameters
+    ----------
+    name : string
+        name of the optic
+    psd_parameters: list of array with various astropy quantities
+        Specifies the various PSD parameters with appropriate units, 
+        ordered according to which PSD model set. Follows the order:
+        [alpha, beta, outer_scale, inner_scale, surf_roughness]
```

Comment:
This one is slightly tricky. We're taking a piecewise approach since a single set of parameters doesn't quite fit well with modeling the measured PSD. This is supposed to be a list of lists, where each list element is a list of the different PSD model parameter quantities. So, if a PSD needs to be modeled by combining 2 PSDs together, then it would show up as` [ [alpha1, beta1, outer_scale1, inner_scale1, surf_roughness1], [alpha2, beta2, outer_scale2, inner_scale2, surf_roughness2]]`
Does this make sense? I'm open for discussing this better so the documentation makes sense.